fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -1178,25 +1184,25 @@ private ScalarOperator getCompensationResidualPredicate(ScalarOperator srcPu,
         } else if (targetPu == null) {
             compensationPu = srcPu;
         } else {
-            ScalarOperator canonizedSrcPu = MvUtils.canonizePredicateForRewrite(srcPu.clone());
-            ScalarOperator canonizedTargetPu = MvUtils.canonizePredicateForRewrite(targetPu.clone());
             ScalarOperator swappedSrcPu;
             ScalarOperator swappedTargetPu;
             if (isQueryAgainstView) {
                 // src is query
-                swappedSrcPu = columnRewriter.rewriteByQueryEc(canonizedSrcPu);
+                swappedSrcPu = columnRewriter.rewriteByQueryEc(srcPu.clone());
```

Comment:
why clone here ? `rewriteByQueryEc ` can change the input?

Reply:
to make sure the swappedSrcPu is new object because the following process will change swappedSrcPu and swappedSrcPu may be null.